### DISCUSSION TITLE
WarpAffine

### DISCUSSION THREAD

**Discussion by AlOa** [Tencent/ncnn#4458] [General]
Hi,
I have some problems with image affine transformation.
I load an image RGB image with  cv::imread and convert to ncnn::Mat with  ncnn::Mat::from_pixels. Then I calculate an affine transformation between to set of points ( I check the transformation with opencv an gives the same affine transformation ). The I try to apply the transformation with warpaffine_bilinear_c3, but the resulting matrix has floating point values, with negative values and some very large value.
Thanks
                        Alberto

**Comment by nihui** [comment]
warpaffine_bilinear_c3 need pixel buffer of uchar type, but from_pixels always convert pixels to float type
so, apply warpaffine before converting to Mat

**Comment by AlOa** [comment]
Now it works, thanks
                               Alberto
